test/functional/apps/dashboard/group5/dashboard_panel_listing.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import expect from '@kbn/expect';
+
+import { FtrProviderContext } from '../../../ftr_provider_context';
+
+export default function ({ getService, getPageObjects }: FtrProviderContext) {
+  const PageObjects = getPageObjects(['dashboard', 'header', 'visualize', 'settings', 'common']);
+  const testSubjects = getService('testSubjects');
+  const kibanaServer = getService('kibanaServer');
+  const dashboardAddPanel = getService('dashboardAddPanel');
+
+  describe('dashboard panel listing', () => {
+    before(async () => {
+      await kibanaServer.savedObjects.cleanStandardList();
+      await kibanaServer.importExport.load(
+        'test/functional/fixtures/kbn_archiver/dashboard/current/kibana'
+      );
+      await kibanaServer.uiSettings.replace({
+        defaultIndex: '0bf35f60-3dc9-11e8-8660-4d65aa086b3c',
+      });
+    });
+
+    after(async () => {
+      await kibanaServer.savedObjects.cleanStandardList();
+    });
+
+    it('renders a panel with predefined order of panel groups', async () => {
+      await PageObjects.dashboard.navigateToApp();
+      await PageObjects.dashboard.clickNewDashboard();
+      await PageObjects.dashboard.switchToEditMode();
+
+      await dashboardAddPanel.clickEditorMenuButton();
+
+      const panelSelectionList = await testSubjects.find('dashboardPanelSelectionList');
+
+      const panelGroupByOrder = new Map();
+
+      const panelGroups = await panelSelectionList.findAllByCssSelector(
+        '[data-test-subj*="dashboardEditorMenu-"]'
+      );
+
+      for (let i = 0; i < panelGroups.length; i++) {
+        const panelGroup = panelGroups[i];
+        const order = await panelGroup.getAttribute('data-group-sort-order');
+        // @ts-ignore
+        const [, panelGroupTitle] = (await panelGroup.getAttribute('data-test-subj'))?.match(
+          /dashboardEditorMenu-(.*)/
+        );
+
+        panelGroupByOrder.set(order, panelGroupTitle);
+      }
+
+      expect([...panelGroupByOrder.values()]).to.eql([
+        'visualizationsGroup',
+        'annotation-and-navigationGroup',
+        'observabilityGroup',
+        'legacyGroup',
+      ]);
+    });
+  });
+}
```

Comment:
This is an interesting case, where we are using functional tests to verify registry values. My concern is that functional tests are becoming such a burden to forward progress. These tests feel like they will add to that burden because they couple the test to dashboard UI. 

I almost wonder if we should test this another way. Instead of dashboard, maybe a better place would be an example application that just renders a very simple page from the registry - then a functional test can just grab this output and verify it's as expected. That way, the test is not coupled to the dashboard UI and instead, is only dependent on the contents of the registry. Thoughts?

Reply:
> These tests feel like they will add to that burden because they couple the test to dashboard UI.

Is there a way to test the registry values outside of a functional test? I thought about it, but obviously stuff like RTL require you to **mock** the registry so it's not testing the "real world" values - to me, this seems like a perfect example of where a functional test might be **necessary** (whereas many of our other tests are... overkill.... and should never have been functional tests in the first place - larger discussion, obviously 😄)

> Instead of using dashboard application, maybe a better place would be an example application that just renders a very simple page from the registry

Hmmm... I think this is an interesting idea. But in this case, is being tied to the dashboard UI really that big of a deal? If all we are doing is loading an empty dashboard, clicking the "add panel" button, and then counting a bunch of stuff.... Is that really that different from navigating to an example plugin and checking counts there? Also, this would introduce overhead - we would have to create the example plugin, maintain it, etc. I think I'm leaning towards keeping this in the Dashboard UI but keeping the tests **as simple as possible** - i.e. checking **counts** and not values.

Reply:
> Hmmm... I think this is an interesting idea. But in this case, is being tied to the dashboard UI really that big of a deal? If all we are doing is loading an empty dashboard, clicking the "add panel" button, and then counting a bunch of stuff.... Is that really that different from navigating to an example plugin and checking counts there? Also, this would introduce overhead - we would have to create the example plugin, maintain it, etc. I think I'm leaning towards keeping this in the Dashboard UI but keeping the tests as simple as possible - i.e. checking counts and not values.

The difference is that a test that relies on dashboard UI will have to be updated if we say, change data-test-subjs or other DOM used by the test. A test that just uses the registry would not need to be updated on UI changes. But yes, it would require the maintaining of an example plugin.

Reply:
So... what is the conclusion here? :) Are we okay to keep the tests in the Dashboard UI for now? Do we want to hold off adding these tests?

Reply:
Adding the test to dashboard UI is ok, we should have a broader discussion about the right place for these types of tests

Reply:
> This is an interesting case, where we are using functional tests to verify registry values. My concern is that functional tests are becoming such a burden to forward progress. These tests feel like they will add to that burden because they couple the test to dashboard UI. 

I do agree ftrs can sometimes be a burden, but I think this is one of those cases where it's entirely useful. To Hannah's point this sort of verification is not something we can exactly do with a unit test.  

> So to me, we could essentially have two tests in this suite:
> -Check that the panel group count is what we expect - i.e. 4
> - Check that the individual panel type count is what we expect - i.e. this would be our "embeddable registry" check, and any new embeddable would just require this value to be incremented (i.e. minimal work for the person making the embeddable, but would still trigger a review from us)
Both of which should be very quick to run and minimally flaky.

I like this, especially that it would also include verifying the panel group items. I however think that verifying the panel grouping render order is still something to tackle in addition to the count.

Reply:
@eokoneyo Would you mind adding this test to this PR? Or would you prefer we create a follow-up issue? 🙇

Reply:
Hey @Heenawter apologies it taken a while to get back to you on this, I think for the time being lets get this one in, I'll create a follow up issue that accounts for adding the tests too verify each group contains the items it should.